llvm/include/llvm/Transforms/IPO/FunctionSpecialization.h

```diff
@@ -252,6 +252,7 @@ class FunctionSpecializer {
   SmallPtrSet<Function *, 32> FullySpecialized;
   DenseMap<Function *, CodeMetrics> FunctionMetrics;
   DenseMap<Function *, unsigned> FunctionGrowth;
+  unsigned NGlobals;
```

Comment:
```suggestion
  unsigned NGlobals = 0;
```

Leave it uninitialized may be a UB.